pkg/util/kubeadm.go

```diff
@@ -38,9 +38,13 @@ func GetKubeadmConfigForMachine(
 	machine *clusterv1.Machine) (*bootstrapv1.KubeadmConfig, error) {
 
 	kubeadmConfig := &bootstrapv1.KubeadmConfig{}
+	kubeadmConfigNamespace := machine.Namespace
+	if machine.Spec.Bootstrap.ConfigRef.Namespace != "" {
```

Comment:
Which one do we want to win out? Is there an interaction with the conversion webhooks here? cc @ncdc

Reply:
I'd say, we need to preserve the user's intent, If not we default to `machine.Namespace`

Reply:
Liggitt suggested yesterday in the context of the CAPI main controllers we explicitly don't use the configref namespace if our intention is to delete it, hence the ask.

It's also an interesting attack vector to steal kubeadmsecrets from namespaces the user might not have access to.

Reply:
We just want to use `machine.Namespace`. We do not allow cross-namespace references. I liked Jordan's suggestion to forcibly reject (validation webhook in CAPI) if the ref's namespace is set, as we want to remove the field entirely in the future.